### ISSUE TITLE
iOS warning: shifting a negative signed value is undefined [-Wshift-negative-value]

### ISSUE THREAD

**Issue by karliwalti** [part-cw/lambdanative#390]
Just started again with building for iOS on MAC

getting this error code for DemoHelloWorld

```
/Users/wkarlen/Library/Caches/lambdanative/ios/arm64/build/2944264680.c:4439:13: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
 ___DEF_CNS(___REF_SYM(12,___S_invalid_2d_clock_2d_type),___REF_CNS(1))
 ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/wkarlen/Library/Caches/lambdanative/ios/arm64/include/gambit.h:2133:48: note: expanded from macro '___REF_SYM'
#define ___REF_SYM(i,id)((___CAST(___WORD,-1-i)<<___TB)+___tMEM2)
                                               ^
/Users/wkarlen/Library/Caches/lambdanative/ios/arm64/include/gambit.h:5449:84: note: expanded from macro '___DEF_CNS'
#define ___DEF_CNS(car,cdr)___MAKE_HD((___PAIR_SIZE<<___LWS),___sPAIR,___PERM),cdr,car
```

**Comment by mgorges** [comment]
Can you please post the actual error - this is a warning, which will not stop compilation and there are lots of them.

**Comment by karliwalti** [comment]
Yes, sorry, the actual error is 

```
2944264680.c:271268:24: error: implicit declaration of function 'getpid' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
___CFUN_CALL(___result,getpid())

```

All I could find about this is here: https://developer.apple.com/forums/thread/127217

**Comment by mgorges** [comment]
`getpid` is defined in [modules/ln_core/log.scm#L137](https://github.com/part-cw/lambdanative/blob/f6dae46ec28e0de104d4abd0bbc0c4d235628118/modules/ln_core/log.scm#L137) - I guess this needs declaring?

**Comment by mgorges** [comment]
Would this be enough?
```
(c-declare  #<<end-of-c-declare
#include <unistd.h>
end-of-c-declare
)
```

**Comment by mgorges** [comment]
Fixed in eabcc49
